File: Anzeige2/src/main/java/com/capgemini/capybench/Anzeige2/dto/PostDto.java
package com.capgemini.capybench.Anzeige2.dto;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;

import java.time.LocalDateTime;
import java.util.Set;

@Value
@Builder
public class PostDto {
    long id;
    @NonNull String title;
    @NonNull String content;
    @NonNull LocalDateTime createDate;
    @NonNull PersonDto author;
    LocalDateTime expirationDate;
    Set<Long> subscriberIds; // if contains ID of the requesting Person, then that Person is subscribing
    long topicId;
    Long subtopicId;
}

File: Anzeige 2.0/src/components/dto/PostDto.ts
import { Person } from "../../domain/Person";
import { Post } from "../../domain/Post";
import { Subtopic } from "../../domain/Subtopic";
import { Topic } from "../../domain/Topic";

class PostDto {
    id: number;
    title: string;
    content: string;
    createDate: Date;
    author: Person;
    expirationDate: Date;
    subscriberIds: number[];
    topicId: number;
    subtopicId: number;

    constructor(id: number, title: string, 
                content: string, createDate: Date, 
                author: Person, expirationDate: Date,
                subsciberIds: number[], topicId: number, 
                subtopicId: number) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.author = author;
        this.createDate = new Date(createDate);
        this.expirationDate = new Date(expirationDate);
        this.subscriberIds = subsciberIds;
        this.topicId = topicId;
        this.subtopicId = subtopicId;
    }

    mapToDomain(topics: Topic[], subtopics: Subtopic[]) : Post {
        return new Post(
            this.id,
            this.title,
            this.content,
            this.createDate,
            this.author,
            this.expirationDate,
            this.subscriberIds,
            topics.find(topic => topic.id === this.topicId)!,
            subtopics.find(subtopic => subtopic.id === this.subtopicId)
        );
    }
}

export { PostDto }